drivers/usb_c/vbus/usbc_vbus_tcpci.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_usb_c_vbus_tcpci
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(usbc_vbus_tcpci, CONFIG_USBC_LOG_LEVEL);
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/usb_c/tcpci_priv.h>
+#include <zephyr/drivers/usb_c/usbc_vbus.h>
+#include <zephyr/drivers/usb_c/usbc_pd.h>
+#include <zephyr/usb_c/tcpci.h>
+
+/** Configuration structure for device instances */
+struct vbus_tcpci_cfg {
+	/** I2C bus and address used for communication, set from parent node of device. */
+	const struct i2c_dt_spec i2c;
+};
+
+static int tcpci_measure(const struct device *dev, int *vbus_meas)
```

Comment:
nit: check if vbus_meas is null and return error if it is

Reply:
I've copied the code from ADC VBUS and used assert for that.